src/kOS.Safe/Execution/Stack.cs

```diff
@@ -61,13 +60,14 @@ private object ProcessItem(object item)
             }
             if (item is double)
             {
-                var outOfBounds = Int32.MinValue < (double) item && (double) item < Int32.MaxValue;
+                var doubleItem = (double)item;
+                var outOfBounds = int.MinValue < doubleItem && doubleItem < int.MaxValue;
 
-                if (!Double.IsNaN((double)item) && outOfBounds)
+                if (!double.IsNaN(doubleItem) && outOfBounds)
                 {
-                    int intPart = Convert.ToInt32(item);
-                    if ((double)item == intPart)
-                        item = intPart;
+                    int intItem = Convert.ToInt32(item);
+                    if (doubleItem == intItem)
+                        item = intItem;
```

Comment:
I caught that too, if you look at the implementation in `ScalarValue.Create` I changed the name to `inBounds`, I did not refactor the method the same way that @erendrake did however.  I'll plan on copying this syntax to my implementation for consistency sake, but it will probably end up deleted from the stack when my PR is ready to merge.